ocaml/libs/http-svr/buf_io.ml

```diff
@@ -79,14 +79,17 @@ let is_full ic = ic.cur = 0 && ic.max = Bytes.length ic.buf
 let fill_buf ~buffered ic timeout =
   let buf_size = Bytes.length ic.buf in
   let fill_no_exc timeout len =
-    let l, _, _ = Unix.select [ic.fd] [] [] timeout in
-    if List.length l <> 0 then (
-      let n = Unix.read ic.fd ic.buf ic.max len in
-      ic.max <- n + ic.max ;
-      if n = 0 && len <> 0 then raise Eof ;
-      n
-    ) else
-      -1
+    Unix.setsockopt_float ic.fd Unix.SO_RCVTIMEO timeout ;
+    let result =
+      try
+        let n = Unix.read ic.fd ic.buf ic.max len in
+        ic.max <- n + ic.max ;
+        if n = 0 && len <> 0 then raise Eof ;
+        n
+      with Unix.Unix_error (Unix.EAGAIN, _, _) -> -1
+    in
+    Unix.setsockopt_float ic.fd Unix.SO_RCVTIMEO 0. ;
```

Comment:
I think we'd better use `finally` to ensure that this timeout is reset in case of any exception (only one kind is caught above.

Reply:
Every `read` on `ic.fd` seems to go through here, and each read is preceded by setting a timeout, so I don't think we even need to reset the timeout, just drop this line.

Reply:
There are more reads in http.ml but they also set a socket timeout. Will need to check the other callers too.